src/hotspot/share/opto/superword.cpp

```diff
@@ -38,146 +38,51 @@
+  _block(arena, 8,  0, nullptr),                            // nodes in current block
+  _data_entry(arena, 8,  0, nullptr),                       // nodes with all inputs from outside
+  _mem_slice_head(arena, 8,  0, nullptr),                   // memory slice heads
+  _mem_slice_tail(arena, 8,  0, nullptr),                   // memory slice tails
+  _node_info(arena, 8,  0, SWNodeInfo::initial),            // info needed per node
+  _clone_map(phase()->C->clone_map()),                      // map of nodes created in cloning
   _align_to_ref(nullptr),                                   // memory reference to align vectors to
-  _dg(_arena),                                              // dependence graph
-  _nlist(arena(), 8, 0, nullptr),                           // scratch list of nodes
-  _lpt(nullptr),                                            // loop tree node
-  _lp(nullptr),                                             // CountedLoopNode
-  _loop_reductions(arena()),                                // reduction nodes in the current loop
-  _bb(nullptr),                                             // basic block
-  _iv(nullptr),                                             // induction var
+  _dg(arena),                                               // dependence graph
+  _nlist(arena, 8, 0, nullptr),                             // scratch list of nodes
+  _loop_reductions(arena),                                  // reduction nodes in the current loop
   _race_possible(false),                                    // cases where SDMU is true
-  _early_return(true),                                      // analysis evaluations routine
-  _do_vector_loop(phase->C->do_vector_loop()),              // whether to do vectorization/simd style
+  _do_vector_loop(phase()->C->do_vector_loop()),            // whether to do vectorization/simd style
   _num_work_vecs(0),                                        // amount of vector work we have
   _num_reductions(0)                                        // amount of reduction work we have
 {
 }
 
-//------------------------------transform_loop---------------------------
-bool SuperWord::transform_loop(IdealLoopTree* lpt, bool do_optimization) {
-  assert(_phase->C->do_superword(), "SuperWord option should be enabled");
-  // SuperWord only works with power of two vector sizes.
-  int vector_width = Matcher::vector_width_in_bytes(T_BYTE);
-  if (vector_width < 2 || !is_power_of_2(vector_width)) {
-    return false;
-  }
-
-  assert(lpt->_head->is_CountedLoop(), "must be");
-  CountedLoopNode *cl = lpt->_head->as_CountedLoop();
-
-  if (!cl->is_valid_counted_loop(T_INT)) {
-    return false; // skip malformed counted loop
-  }
-
-  // Initialize simple data used by reduction marking early.
-  set_lpt(lpt);
-  set_lp(cl);
-  // For now, define one block which is the entire loop body.
-  set_bb(cl);
-
-  if (SuperWordReductions) {
-    mark_reductions();
-  }
```

Comment:
Note: I now would like to move reduction marking to **after** precondition checking. Hence, I moved it to `SLP_extract`.